lib/src/model.dart

```diff
@@ -3718,29 +3741,69 @@ abstract class ModelElement extends Canonicalization
     });
   }
 
+  /// Helper to process arguments given as a (possibly quoted) string.
+  ///
+  /// First, this will split the given [argsAsString] into separate arguments,
+  /// taking any quoting (either ' or " are accepted) into account, including
+  /// handling backslash-escaped quotes.
+  ///
+  /// Then, it will prepend "--" to any args that start with an identifier
+  /// followed by an equals sign, allowing the argument parser to treat any
+  /// "foo=bar" argument as "--foo=bar". It does handle quoted args like
+  /// "foo='bar baz'" too, returning just bar (without quotes) for the foo
+  /// value.
+  ///
+  /// It then parses the resulting argument list normally with [argParser] and
+  /// returns the result.
+  ArgResults _parseArgs(String argsAsString, ArgParser argParser) {
+    // Regexp to take care of splitting arguments, and handling the quotes
+    // around arguments, if any.
+    //
+    // Match group 1 is the "foo=" part of the option, if any.
+    // Match group 2 contains the quote character used (which is discarded).
+    // Match group 3 is a quoted arg, if any, without the quotes.
+    // Match group 4 is the unquoted arg, if any.
+    final RegExp argMatcher = new RegExp(r'([a-zA-Z\-_0-9]+=)?' // option name
+        r'(?:' // Start a new non-capture group for the two possibilities.
+        r'''(["'])((?:\\{2})*|(?:.*?[^\\](?:\\{2})*))\2|''' // with quotes.
+        r'([^ ]+))'); // without quotes.
+    final Iterable<Match> matches = argMatcher.allMatches(argsAsString);
+
+    // Remove quotes around args, and for any args that look like assignments
+    // (start with valid option names followed by an equals sign), add a "--" in front
+    // so that they parse as options.
+    final Iterable<String> args = matches.map<String>((Match match) {
+      String option = '';
+      if (match[1] != null) {
+        option = '--${match[1]}';
+      }
+      return option + (match[3] ?? '') + (match[4] ?? '');
+    });
+
+    return argParser.parse(args);
```

Comment:
I think ArgParserExceptions thrown here probably should not be fatal and should instead emit some sort of helpful warning.  You might have to carry along some extra data into _parseArgs to make that useful without reflection.